File: app/src/main/java/com/boardgamegeek/sorter/PlayCountSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.database.Cursor;
import android.support.annotation.NonNull;

import com.boardgamegeek.R;
import com.boardgamegeek.provider.BggContract.Collection;

public abstract class PlayCountSorter extends CollectionSorter {
	public PlayCountSorter(@NonNull Context context) {
		super(context);
	}

	@Override
	public String[] getColumns() {
		return new String[] { Collection.NUM_PLAYS };
	}

	@Override
	public String getHeaderText(@NonNull Cursor cursor) {
		return getIntAsString(cursor, Collection.NUM_PLAYS, "0");
	}

	@NonNull
	@Override
	public String getDisplayInfo(@NonNull Cursor cursor) {
		return getHeaderText(cursor) + " " + context.getString(R.string.plays);
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/PlayCountSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes
import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItem
import java.text.NumberFormat

class PlayCountSorter(context: Context) : CollectionSorter(context) {
    override val ascendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_play_count_asc

    override val descendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_play_count_desc

    override val descriptionResId: Int
        @StringRes
        get() = R.string.collection_sort_play_count

    override fun sortAscending(items: Iterable<CollectionItem>) = items.sortedBy { it.numberOfPlays }

    override fun sortDescending(items: Iterable<CollectionItem>) = items.sortedByDescending { it.numberOfPlays }

    override fun getHeaderText(item: CollectionItem): String = NumberFormat.getIntegerInstance().format(item.numberOfPlays)

    override fun getDisplayInfo(item: CollectionItem) =
        context.resources.getQuantityString(R.plurals.plays, item.numberOfPlays, getHeaderText(item))
}
